indico/modules/events/papers/client/js/components/PaperReviewForm.jsx

```diff
@@ -5,8 +5,111 @@
+
+import {addComment} from '../actions';
+import {canCommentPaper, getPaperDetails, getCurrentUser, isAddingComment} from '../selectors';
+import UserAvatar from './UserAvatar';
+
+import './PaperReviewForm.module.scss';
 
 export default function PaperReviewForm() {
-  return <div>TODO: Review Form</div>;
+  const {
+    event: {id: eventId},
+    contribution: {id: contributionId},
+  } = useSelector(getPaperDetails);
+  const user = useSelector(getCurrentUser);
+  const canComment = useSelector(canCommentPaper);
+  const isCommentBeingAdded = useSelector(isAddingComment);
+  const dispatch = useDispatch();
+  const [commentFormVisible, setCommentFormVisible] = useState(false);
+
+  const visibilityOptions = [
+    {
+      value: 'judges',
+      text: Translate.string('Visible only to judges'),
+    },
+    {
+      value: 'reviewers',
+      text: Translate.string('Visible to reviewers and judges'),
+    },
+    {
+      value: 'contributors',
+      text: Translate.string('Visible to contributors, reviewers and judges'),
+    },
+  ];
+
+  const onCommentClickHandler = commentFormVisible
+    ? null
+    : () => {
+        setCommentFormVisible(true);
+      };
+
+  const createComment = useCallback(
+    formData => {
+      dispatch(addComment(eventId, contributionId, formData));
+    },
+    [dispatch, eventId, contributionId]
+  );
+
+  return (
+    <div className="i-timeline-item" styleName="review-timeline-input">
+      <UserAvatar user={user} />
+      <div className="i-timeline-item-box footer-only header-indicator-left">
+        <div className="i-box-footer" style={{overflow: 'visible'}}>
+          {canComment && (
+            <div className="flexrow">
+              <div className="f-self-stretch">
+                <FinalForm
+                  onSubmit={createComment}
+                  initialValues={{comment: '', visibility: 'judges'}}
+                >
+                  {fprops => (
+                    <Form
+                      id="comment-form"
+                      onSubmit={fprops.handleSubmit}
+                      loading={isCommentBeingAdded}
+                    >
+                      <FinalTextArea
+                        onFocus={onCommentClickHandler}
+                        name="comment"
+                        rows={commentFormVisible ? 3 : 1}
+                        placeholder={Translate.string('Leave a comment...')}
+                      />
+                      {commentFormVisible && (
+                        <>
+                          <FinalDropdown
+                            name="visibility"
+                            options={visibilityOptions}
+                            selection
+                            required
+                          />
+                          <Form.Group inline>
+                            <FinalSubmitButton
+                              label={Translate.string('Comment')}
+                              form="comment-form"
```

Comment:
we're inside the form so this is most likely not needed (neither is the `id` on the form in this case)